trino/client.py

```diff
@@ -803,7 +803,10 @@ def execute(self, additional_http_headers=None) -> TrinoResult:
         if self.cancelled:
             raise exceptions.TrinoUserError("Query has been cancelled", self.query_id)
 
-        response = self._request.post(self._query, additional_http_headers)
+        try:
+            response = self._request.post(self._query, additional_http_headers)
+        except requests.exceptions.RequestException as e:
+            raise trino.exceptions.TrinoConnectionError("failed to execute: {}".format(e))
```

Comment:
I think a [OperationalError](https://peps.python.org/pep-0249/#operationalerror) is more at its place here.

This also applies to the other places where you wrap this exception, otherwise it would not be captured by other frameworks.